packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-inline-checkout-button-strategy.ts

```diff
@@ -0,0 +1,334 @@
+
+        return orderId;
+    }
+
+    private async _onShippingAddressChange(data: ShippingAddressChangeCallbackPayload): Promise<void> {
+        const address = this._transformAddress({
+            city: data.shippingAddress.city,
+            countryCode: data.shippingAddress.country_code,
+            postalCode: data.shippingAddress.postal_code,
+            stateOrProvinceCode: data.shippingAddress.state,
+        });
+
+        // Info: we use the same address to fill billing and consignment addresses to have valid quota on BE for order updating process
+        // on this stage we don't have access to valid customer's address accept shipping data
+        await this._store.dispatch(this._billingAddressActionCreator.updateAddress(address));
+        await this._store.dispatch(this._consignmentActionCreator.updateAddress(address));
+
+        const shippingOption = this._getShippingOptionOrThrow();
+
+        await this._store.dispatch(this._consignmentActionCreator.selectShippingOption(shippingOption.id));
+        await this._updateOrder();
+    }
+
+    private async _onShippingOptionsChange(data: ShippingOptionChangeCallbackPayload): Promise<void> {
+        const shippingOption = this._getShippingOptionOrThrow(data.selectedShippingOption?.id);
+
+        await this._store.dispatch(this._consignmentActionCreator.selectShippingOption(shippingOption.id));
+        await this._updateOrder();
+    }
+
+    private async _onApprove(
+        data: ApproveCallbackPayload,
+        actions: ApproveCallbackActions,
+        methodId: string,
+    ): Promise<boolean> {
+        const state = this._store.getState();
+        const cart = state.cart.getCartOrThrow();
+        const orderDetails = await actions.order.get();
+
+        if (cart.lineItems.physicalItems.length > 0) {
+            const address = this._getValidAddress(
+                orderDetails.payer.name,
+                orderDetails.payer.email_address,
+                orderDetails.purchase_units[0].shipping.address
+            );
+
+            await this._store.dispatch(this._billingAddressActionCreator.updateAddress(address));
+            await this._store.dispatch(this._consignmentActionCreator.updateAddress(address));
+            await this._updateOrder();
+        } else {
+            const address = this._getValidAddress(
+                orderDetails.payer.name,
+                orderDetails.payer.email_address,
+                orderDetails.payer.address
+            );
+
+            await this._store.dispatch(this._billingAddressActionCreator.updateAddress(address));
+        }
+
+        await this._store.dispatch(this._orderActionCreator.submitOrder({}, { params: { methodId } }));
+        await this._submitPayment(methodId, data.orderID);
+
+        return true;
+    }
+
+    private async _onComplete(
+        data: CompleteCallbackDataPayload,
+        methodId: string,
+        callback?: () => void
```

Comment:
```suggestion
        callback?(): void
```

Reply:
I can't make it for function argument due to typescript error:
<img width="491" alt="Screenshot 2022-09-07 at 15 06 15" src="https://user-images.githubusercontent.com/25133454/188874567-b00a0cac-8428-4af4-8dc5-780095ca09c0.png">
<img width="612" alt="Screenshot 2022-09-07 at 15 06 42" src="https://user-images.githubusercontent.com/25133454/188874579-81fb3621-5d7b-469f-ac6e-ee57a248e9ea.png">